Sources/ArgumentParser/Parsable Types/ParsableCommand.swift

```diff
@@ -164,4 +171,19 @@ extension ParsableCommand {
   internal static var defaultIncludesUnconditionalArguments: Bool {
     configuration.defaultSubcommand?.includesUnconditionalArguments == true
   }
+  
+#if DEBUG
+  @available(macOS 10.15, macCatalyst 13, iOS 13, tvOS 13, watchOS 6, *)
+  internal static func checkAsyncHierarchy(_ command: ParsableCommand.Type) {
+    command.configuration.subcommands.forEach(checkAsyncHierarchy)
+    
+    if command is AsyncParsableCommand.Type { return }
+    
+    for sub in command.configuration.subcommands {
+      guard !(sub is AsyncParsableCommand.Type) else {
+        fatalError("`\(sub)` subcommand can't run asynchronously in `\(command)` command")
```

Comment:
I don't think this is giving quite the right diagnostic — if the root and first-level subcommand are synchronous, and the second-level subcommand is async, it will point at the first-level subcommand instead of the root.

```swift
@main
struct Root: ParsableCommand {
  static var configuration: CommandConfiguration {
    .init(subcommands: [SubOne.self])
  }

  struct SubOne: ParsableCommand {
    static var configuration: CommandConfiguration {
      .init(subcommands: [SubTwo.self])
    }
    
    struct SubTwo: AsyncParsableCommand {}
  }
}
```

Gives this error even though `Root` is the problem, not `SubOne`:

```
Fatal error: `SubTwo` subcommand can't run asynchronously in `SubOne` command
```

Maybe you can do a recursive search for any `AsyncParseableCommand`, and then output the error message just from this root-level context.

Reply:
Additionally, let's update that error message to include information about how to fix the problem. Since this is a configuration issue, it would be great if this could be as big and clear a message as possible. Let's borrow the format from `directlyInitializedError`:

```
fatalError("""

  --------------------------------------------------------------------
  Asynchronous subcommand of a synchronous root.

  The asynchronous command `\(sub)` is declared as a subcommand of the synchronous root command `\(command)`.
  
  With this configuration, your asynchronous `run()` method will not be called. To fix this issue, change `\(command)`'s `ParsableCommand` conformance to `AsyncParsableCommand`.
  --------------------------------------------------------------------

  """.wrapped(to: 70))
```

Reply:
done